CC/include/SquareMatrix.h

```diff
@@ -115,16 +115,14 @@ namespace CCLib
 		**/
 		void invalidate()
 		{
-			if (m_values)
-			{
-				for (unsigned i = 0; i < m_matrixSize; i++)
-					if (m_values[i])
-						delete[] m_values[i];
-				delete[] m_values;
-				m_values = nullptr;
-			}
+			delete [] m_underlyingData;
```

Comment:
I'm not sure I follow. "valid" in what sense?

Deleting a `nullptr` has no effect.

Reply:
Deleting a nullptr makes the program crash! (at least on my side?!)

Reply:
That means we have a bigger problem! :-) 

Deleting `nullptr` is allowed by the standard, so either there's a bug lurking here somewhere or MSVC is broken (unlikely).

>For the second (array) form, expression must be a null pointer value or a pointer value previously obtained by an array form of new-expression. If expression is anything else, including if it's a pointer obtained by the non-array form of new-expression, the behavior is undefined.

Reply:
Is it a new behavior? (with the introduction of nullptr?). It's a long time since I don't try to do that anymore :D

Reply:
Nope - always been part of C++.

https://en.cppreference.com/w/cpp/language/delete

> If expression evaluates to a null pointer value, no destructors are called, and the deallocation function may or may not be called (it's implementation-defined), but the default deallocation functions are guaranteed to do nothing when handed a null pointer. | (until C++14)

Even in C you could `free( NULL )`.

Reply:
Ok ... I should try again then! I'm pretty sure it was not the case with Visual Studio back in the old days ;)

Reply:
What's the stack trace of the crash? Is the pointer `null` right before the delete?

I'm not seeing that we're doing anything strange with that pointer or data here.

Reply:
No, it's not crashing here. I meant that as a general rule (as I said I never write stuff like this since I learned to test a pointer before deleting it a long time ago).